synapse/storage/databases/main/relations.py

```diff
@@ -431,17 +433,33 @@ def _get_thread_summary_txn(
             txn.execute(sql, (event_id, room_id, RelationTypes.THREAD))
             count = cast(Tuple[int], txn.fetchone())[0]
 
-            return count, latest_event_id
+            # Fetch whether the requester has participated or not.
+            sql = """
+                SELECT 1
+                FROM event_relations
+                INNER JOIN events USING (event_id)
+                WHERE
+                    relates_to_id = ?
```

Comment:
Sounds good to me.

As for the indices, I think the existing one is sufficient, because a __btree__ index on (X, Y, Z) can be used as an index on (X, Y). But I can't find a reference for this. Maybe @reivilibre knows one?

Reply:
> But I can't find a reference for this.

Searching for "Index" and "prefix" gets you stuff related to `WHERE strcol LIKE 'foo%'`. What I wanted to find was

https://use-the-index-luke.com/sql/where-clause/the-equals-operator/concatenated-keys and https://www.postgresql.org/docs/10/indexes-multicolumn.html . To quote from postgres docs specifically

> A multicolumn B-tree index can be used with query conditions that involve any subset of the index's columns, but the index is most efficient when there are constraints on the leading (leftmost) columns.